### PR TITLE
adding tests to check the withdraw function

### PR DESCRIPTION
**Pull request by faytey**

### CODE REVIEW

**Review by sprtd — COMMENTED** [review]
Good job, Faith. Your `Withdrawal` test case is passing however, there are a few observations:
1. [L84-92](https://github.com/faytey/intro-to-testing/blob/faith_test/test/ColabBank.test.js#L84-92) should be enclosed within the preceeding `it` block
2. Your `Events` test case is referencing a different contract `Lock`. You should be fine if you correctly reference `colabBank` contract in this test
You are eligible for the reward when you fix these

Again, there's a bug in this contract. Can you spot it? You'll get a bonus if you report it

Overall, you did well 👍🏿

**Comment by faytey** [comment]
Thanks for the feedback, i don't seem to understand the corrections please,
my L84-92 is enclosed in the it block, or do you mean I should create a
different block for it, also I can't seem to understand where I referenced
lock contract instead of colabbank contract, I've checked even the event
arguments are (amount and when) and that's why I used (3 and anyvalue),
please I don't understand the correction please

**Comment by sprtd** [comment]
> Thanks for the feedback, i don't seem to understand the corrections please, my L84-92 is enclosed in the it block, or do you mean I should create a different block for it, also I can't seem to understand where I referenced lock contract instead of colabbank contract, I've checked even the event arguments are (amount and when) and that's why I used (3 and anyvalue), please I don't understand the correction please
> […](#)



1. For `Withdraw` test case,  cut [L84-92](https://github.com/faytey/intro-to-testing/blob/faith_test/test/ColabBank.test.js#L84-L92), make sure it is contained within the `it` block. You don't need a separate `it` block for this. use the one you have already

2. For the `Event` test case, in [L98](https://github.com/faytey/intro-to-testing/blob/faith_test/test/ColabBank.test.js#L984), you are extracting/referencing `lock`, which is inaccurate. Target `colabBank` and you should be fine.

**Review by sprtd — COMMENTED** [review]
Great job @faytey. You've effected the changes, now the tests are passing. 

Can you test to check `owner`'s ETH balance before and after withdrawal? You can use `ethers.provider.getBalance`  to get the ETH balance of the `owner` and `colabBank` contract. For details, refer to this [example](https://ethereum.stackexchange.com/questions/96372/how-to-get-contracts-ether-balance-at-hardhat-waffle)

Bonus point: can you spot the vulnerability in this contract? 

Kudos 👍🏿

**Comment by faytey** [comment]
oh wow thanks, i wish i could see the tests passing myself, i'd attempt the
rest tomorrow morning please.

**Comment by sprtd** [comment]
> oh wow thanks, i wish i could see the tests passing myself, i'd attempt the rest tomorrow morning please.
no worries, take your time

**Comment by sprtd** [comment]
> oh wow thanks, i wish i could see the tests passing myself, i'd attempt the rest tomorrow morning please.


no worries, take your time

File: test/ColabBank.test.js [hunk 1]
```diff
@@ -69,14 +66,14 @@ describe("ColabBank Test Suite", async () => {
     })
    // Adding the withdrawal test (Faith M. Roberts)
    describe("Withdrawal", async () => {
-    it("should fail if called before the present time", async () => {
-    const { colabBank, addr1, unlockTime, owner } = await loadFixture(deployOneYearLockFixture);
+     it("should fail if called before the present time", async () => {
```
Comment (sprtd) [inline test/ColabBank.test.js]: add an appropriate description for the withdrawal unit test. `should fail if called before the present time` is not appropriate in this context

File: contracts/v2/ColabBankV2.sol [hunk 2]
```diff
@@ -18,7 +18,7 @@ contract ColabBankV2 {
     function deposit() public payable {
         require(msg.value != 0, "cannot deposit 0 ETH");
         uint256 value = msg.value;
-        balances[msg.sender] = value;
+        balances[msg.sender] += value;
```
Comment (sprtd) [inline contracts/v2/ColabBankV2.sol]: I'd suggest you finalize `ColabBank` first before proposing changes to `ColabBankV2`. nice attempt tho

File: test/ColabBank.test.js [hunk 3]
```diff
@@ -51,27 +48,74 @@ describe("ColabBank Test Suite", async () => {
...
             
             const totalColabBalance = await colabBank.totalColabBalance()
-            console.log("total colab balance__", totalColabBalance)
             expect(totalColabBalance).to.eq(5)
 
             await expect(addr1DepositTxn).to.emit(colabBank, "Deposit").withArgs(5, anyValue, addr1.address)
+        });
+    });
+   // Adding the withdrawal test (Faith M. Roberts)
+    describe("Withdrawal", async () => {
```
Comment (sprtd) [inline test/ColabBank.test.js]: Good job further splitting your tests

File: contracts/v2/observations.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# My Observations on the vulnerabilities present in the ColabBank V2.
```
Comment (sprtd) [inline contracts/v2/observations.md:1]: Awesome! You spotted the flawed logic in this contract. Great job 👍🏿

**Review by sprtd — COMMENTED** [review]
You did well. Keep going

**Comment by sprtd** [comment]
nice attempt Faith

**Review by sprtd — APPROVED** [review]

**Comment by faytey** [comment]
Thank you so much

**Review by sprtd — APPROVED** [review]

**Pull request merged**